CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticConfigurationPane.java

```diff
@@ -490,15 +510,29 @@ public static final SelectableAnalyticPlugin lookupSelectablePlugin(final String
         return NAME_TO_SELECTABLE_PLUGIN_MAP.get(selectableAnalyticPluginName);
     }
 
+    private static void setSuppressedFlag(final boolean newValue) {
+        selectionSuppressed = newValue;
+    }
+
     public final class SelectableAnalyticPlugin {
 
         private final CheckBox checkbox;
+        private ListCell<SelectableAnalyticPlugin> parent;
         private final AnalyticPlugin plugin;
         private PluginParameters parameters;
         private PluginParameters updatedParameters;
 
         public SelectableAnalyticPlugin(final AnalyticPlugin plugin) {
             this.checkbox = new CheckBox();
+            // Allows triggering of selection listener when a checkbox is changed
```

Comment:
What is happening here?

Reply:
https://github.com/constellation-app/constellation/blob/fbce6de64cf65397f7d6065cb58cd2b85b3c8e2c/CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticConfigurationPane.java#L527-L535
So, for an event to fire on the selection model of the pluginList, the plugin has to change state to highlighted. If the plugin was already highlighted but did not have it's checkbox "checked", a user could select the checkbox and no update event would occur to store the state.
By adding a listener to the checkbox, I could trigger an event in it's parent (Listviewcell)'s selection model event. I felt that this would be the cleanest way rather than creating another public method to call once the checkbox event was called.

Reply:
Cool - that makes sense.